integration-tests/test_register.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import functools
+import logging
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    response = insights_client.run("--register", check=False)
+    assert (
+        response.returncode == 0
+    ), "An application returns 0 after successful registration"
+    assert (
+        "Successfully registered host" in response.stdout
+    ), "An application should inform that a system was registered"
+    assert (
+        "Starting to collect Insights data" in response.stdout
+    ), "An application should inform that Insights data will be collected"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
+
+
+def test_register_twice(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    insights_client.run("--register")
+    assert insights_client.is_registered
+
+    response = insights_client.run("--register")
+    assert (
+        "This host has already been registered" in response.stdout
+    ), "An application should inform that a system was already registered"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
+
+
+def test_register_using_noauth_proxy(insights_client, test_config):
+    proxy_config = functools.partial(test_config.get, "noauth_proxy")
+    proxy = f'http://{proxy_config("host")}:{proxy_config("port")}'
```

Comment:
I use http schema only. I think we do not have any test using https proxy anymore.

Reply:
my point is that, if you need to use an https proxy for the tests (because it is configured as such), then you need a way to make the tests work with it